pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -156,10 +235,15 @@ def __init__(self, *args, **kargs):
         """
         GraphicsObject.__init__(self)
         self.setFlag(self.GraphicsItemFlag.ItemHasNoContents)
-        self.xData = None
-        self.yData = None
-        self.xDisp = None
-        self.yDisp = None
+        # self.xData = None # original data
+        # self.yData = None
+        self.dataset       = None # will hold a PlotDataset for the original data
+        self.datasetMapped = None # will hold a PlotDataset for data after mapping transforms (e.g. log scale)
+        self.datasetDisp   = None # will hold a PlotDataset for data downsampled and limited for display
```

Comment:
nit-pick level 1001 

I'm trying to avoid introducing new variable names that are abbreviated; may I suggest renaming to `datasetDisplay` ?

Also, these 3 datasets are closely coupled, I keep wondering if they should be members of a dictionary... but I'm not strongly opinionated here.

Reply:
--> changed to datasetDisplay.

I was trying not to jump the degree of encapsulation from zero to max in a single step. I think the individual properties are a little easier to track (compared to stuffing them all into one dictionary) while the refactored structures are still somewhat fragile. I think these are not really user-facing, and we should still be able to clean them up later.
--> renamed to "private" namespace as `_dataset`, `_datasetMapped` and `_datasetDisplay`